### ISSUE TITLE
Flush write-ahead log before push and verify that no WAL files are committed

### ISSUE THREAD

**Issue by pchiusano** [unisonweb/unison#2063]
According to https://sqlite.org/tempfiles.html the WAL files aren’t necessarily flushed on close if there are other open connections to the database. Also see https://github.com/groue/GRDB.swift/issues/418

We are probably not closing the connection on `pull` and that should be tracked down, but we should make the `push` code robust to concurrent connections being open.

Also, we should ensure that `push` can’t commit WAL files or really any files other than the database files. 

```
$ git add v2/sqlite3.db
$ git commit -m blah
$ git status --porcelain # make sure this is empty
```
WIP in fix/wal-flush branch

Arya suggested enabling the WAL on `push` as well.

**Comment by pchiusano** [comment]
We suspect this has been fixed by #2064 but didn't have a reliable repro. Please reopen if it surfaces again!

**Comment by groue** [comment]
Hello @pchiusano, since you refer to GRDB, please let me chime in to say that the removal of WAL files is also controlled by the [SQLITE_FCNTL_PERSIST_WAL](https://sqlite.org/c3ref/c_fcntl_begin_atomic_write.html#sqlitefcntlpersistwal) flag.

Some SQLite distributions mess with this flag, and you may want to explicitly set it to the needed value.

Also, remember that when there is no -wal and -shm files, it is impossible to open a read-only connection to a WAL database. [Reference](https://www.sqlite.org/walformat.html#operations_that_require_locks_and_which_locks_those_operations_use):

> If the WAL and WAL-index files are missing, then a process that lacks permission to create and initialize those files will not be able to open the database, unless the database is designated as immutable using the immutable query parameter.
